types/src/block.rs

```diff
@@ -139,11 +140,13 @@ impl CommitExt for Commit {
             block_id: Some(self.block_id),
             timestamp: Some(timestamp),
             validator_address,
-            validator_index: signature_idx.try_into()?,
+            validator_index: signature_idx.try_into().map_err(Error::Tendermint)?,
```

Comment:
`map_err` is not needed because you implemented the `From`. Remove all `map_err` that you added.